### PR TITLE
comprehensive list of lints

### PR DESCRIPTION
**Pull request by a14n**
All lints from linter-0.1.30 are now listed.

### CODE REVIEW

**Comment by Hixie** [comment]
For the ones that are commented out, can you add a brief comment after them saying why? (some already have one)

**Comment by Hixie** [comment]
@a14n Anything we can do to help you with this patch?

**Comment by a14n** [comment]
oops, I forgot this PR. Thanks for the reminder :)

File: packages/flutter/lib/src/rendering/object.dart [hunk 1]
```diff
@@ -1781,7 +1781,6 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
       final RenderObject parent = this.parent;
       relayoutBoundary = parent._relayoutBoundary;
     }
-    assert(parent == this.parent);
```
Comment (a14n) [inline packages/flutter/lib/src/rendering/object.dart]: I dropped those 2 asserts but perhaps something better should be done.

**Comment by a14n** [comment]
PTAL

File: packages/flutter/lib/src/rendering/box.dart [hunk 2]
```diff
@@ -1769,7 +1769,7 @@ abstract class RenderBox extends RenderObject {
         throw new FlutterError(
           '$runtimeType does not implement applyPaintTransform.\n'
           'The following $runtimeType object:\n'
-          '  ${this.toStringShallow()}\n'
+          '  ${toStringShallow()}\n'
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/box.dart]: Not sure how I feel about this one. I'd intentionally put in the "this" to match the "child" case below.
I guess I'll live with it.

File: packages/flutter/lib/src/rendering/object.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/rendering/object.dart]: It was supposed to be testing the "parent" from the block just before this line. It's fine to remove.

**Comment by Hixie** [comment]
LGTM

...but somehow `flutter analyze` and `flutter analyze --watch` got different results.

**Comment by a14n** [comment]
Not enough knowledge to fix the issue that `flutter analyze --flutter-repo` and `flutter analyze --flutter-repo --watch` do not return the same results. @pq could you have a look ? You should see the issue if you revert 25a9c9bc9b7f89430309d636714bdab0d98d387e

**Comment by a14n** [comment]
or @danrubel (if I trust `git blame`)

**Comment by pq** [comment]
Ugh, yeah.  There's a skew between the version of the linter being pulled into flutter_tools and used in `flutter analyze` and the one in the SDK (and used in `--watch`).  I expect that's what's up.  The SDK is now up to date but we'll have to live with this skew until the first `1.24` dev builds.  This confusion is a drag but I hope short-lived.

**Comment by a14n** [comment]
@pq thanks

File: .analysis_options_repo [hunk 3]
```diff
@@ -67,35 +69,41 @@ linter:
...
+    # - prefer_expression_function_bodies # not yet tested
+    # - prefer_final_fields # https://github.com/dart-lang/linter/issues/506
     - prefer_final_locals
+    # - prefer_function_declarations_over_variables # not yet tested
     - prefer_initializing_formals
+    # - prefer_interpolation_to_compose_strings # not yet tested
     - prefer_is_empty
     - prefer_is_not_empty
-    - public_member_api_docs # this is the only difference from .analysis_options
+    # - public_member_api_docs # this is the only difference from .analysis_options_repo
```
Comment (Hixie) [inline .analysis_options_repo]: I missed this when reviewing it (and will fix it momentarily), but just FYI, this specific line broke `--dartdocs` in `flutter analyze`. :-)

Comment (a14n) [inline .analysis_options_repo]: Oops. Sorry :-/

Comment (Hixie) [inline .analysis_options_repo]: Not a problem, I just wanted to let you know in case you edit that file again. :-)

**Pull request merged**